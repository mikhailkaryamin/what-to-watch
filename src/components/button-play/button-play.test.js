import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';

import renderer from 'react-test-renderer';

import ButtonPlay from './button-play.tsx';

describe(`button play`, () => {
  const FILM = {
    backgroundImage: `img/aviator.jpg`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
    director: `Arthur Agee`,
    genre: `Comedy`,
    id: 1,
    isFavorite: false,
    name: `Devin Albert`,
    posterImage: `img/aviator.jpg`,
    previewImage: `img/aviator.jpg`,
    previewVideoLink: `link`,
    rating: 3,
    released: 2015,
    runTime: 100,
    scoreCount: 300,
    starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
    video: `some/link`,
  };

  const wrapper = renderer
    .create(
        <Router>
          <ButtonPlay
            film={FILM}
          />
        </Router>
    ).toJSON();

  test(`should render button play`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
