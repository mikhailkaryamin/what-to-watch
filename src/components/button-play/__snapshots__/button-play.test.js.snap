// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button play should render button play 1`] = `
<a
  className="btn btn--play movie-card__button"
  href="/player/3"
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  <svg
    height="19"
    viewBox="0 0 19 19"
    width="19"
  >
    <use
      xlinkHref="#play-s"
    />
  </svg>
  <span>
    Play
  </span>
</a>
`;
